lib/Fmt_ast.ml

```diff
@@ -4448,7 +4448,11 @@ let fmt_toplevel ?(force_semisemi = false) c ctx itms =
     let last = Option.is_none next in
     let semisemi =
       match (itm, next) with
+      | `Item {pstr_desc= Pstr_attribute _; _}, _ -> false
       | _, Some (`Item {pstr_desc= Pstr_eval _; _}, _) -> true
+      | ( `Item {pstr_desc= Pstr_eval _; _}
+        , Some (`Item {pstr_desc= Pstr_attribute _; _}, _) ) ->
+          true
```

Comment:
There's no `;;` here:

```ocaml
x + 1

let y = 2
```

Is it really needed here ?

```ocaml
x + 1;;

(** foo *)

let y = 2
```

The diffs in test_branch seems to indicate that no.

Reply:
I fixed the regressions, let me know what you think.